generator/internal/codegen_utils.cc

```diff
@@ -87,6 +88,44 @@ std::string ProtoNameToCppName(absl::string_view proto_name) {
   return "::" + absl::StrReplaceAll(proto_name, {{".", "::"}});
 }
 
+std::vector<std::string> BuildNamespaces(
+    std::map<std::string, std::string> const& vars, NamespaceType ns_type) {
+  // vars["product_path"] is guaranteed to be present and properly formatted.
```

Comment:
What guarantees this? It makes me uncomfortable to have maps with keys that are "guaranteed to exist". If the key must exist, should we use a struct or something with a field that must always exist?